packages/transactions/src/coders/transaction.ts

```diff
@@ -63,7 +63,7 @@ export class TransactionScriptCoder extends Coder {
     parts.push(new NumberCoder('gasPrice', 'u64').encode(value.gasPrice));
     parts.push(new NumberCoder('gasLimit', 'u64').encode(value.gasLimit));
     parts.push(new NumberCoder('bytePrice', 'u64').encode(value.bytePrice));
-    parts.push(new NumberCoder('maturity', 'u32').encode(value.maturity));
+    parts.push(new NumberCoder('maturity', 'u64').encode(value.maturity));
```

Comment:
I think we should keep it as spec even if the decoder results in the same 8 bytes. https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/tx_format.md

Reply:
Hmm. I thought we changed maturity to u64. There are other places where we type maturity u64 so I've opened an issue to fix this completely: https://github.com/FuelLabs/fuels-ts/issues/275

Let's keep this change though so at least all maturities are u64 instead of only one being u32.

Reply:
Actually, this fixes a bug and we need it because the JS type is `bigint` and this returns `number`.

Reply:
cc: @adlerjohn

Reply:
https://github.com/FuelLabs/fuels-ts/issues/275#issuecomment-1127022049